### DISCUSSION TITLE
can I test if the microphone is currently recording with cypress? if I do how?

### DISCUSSION THREAD

**Discussion by ofekOfir** [cypress-io/cypress#27199] [Questions and Help]
I have developed a react component that contains a button that start recording and stop recording an audio I can i identify with cypress when the recording process is on/off?

**Comment by portyu9** [comment]
Hi @ofekOfir 

You can test whether **your application believes it is recording**, but Cypress does not have an API that asks the operating system whether the physical microphone is currently recording.

The best test is against application-observable behavior.

For example, if your UI has a recording indicator:

```js
cy.get('[data-cy=start-recording]').click()

cy.get('[data-cy=recording-indicator]')
  .should('be.visible')

cy.get('[data-cy=stop-recording]').click()

cy.get('[data-cy=recording-indicator]')
  .should('not.exist')
```

For a component test, an even cleaner approach is to put `getUserMedia` / `MediaRecorder` behind an application service and stub that service. Then assert that your component calls `start()` and `stop()` and displays the correct state.

One important detail: Cypress currently launches Chromium with fake-media-device flags by default for WebRTC testing. That means an automated Cypress test should not be interpreted as proof that the user's **real hardware microphone** was captured.

If what you want to test is your application's recording workflow, that fake/deterministic media environment is actually preferable because it is reproducible in CI.

The full discussion was reviewed; there were no screenshots/attachments or replies.

Please mark this answer as accepted if it helped, thank you.
